### ISSUE TITLE
Reopened: Add `sign-git-commit` config for GPG-signed commits during `npm version`

### ISSUE THREAD

**Issue by tribou** [npm/npm#17429]
Reopened #12696 

#### What's the feature?

Original issue:

Currently, `npm@3.8.9` supports a `sign-git-tag` config which enables the `git tag -s` command instead of `git tag` in order to perform GPG-signed tags.  However, the git commit performed prior to the tagging and after the package version is bumped does not support signing.  This results in a signed tag but an unsigned git commit.

For consistency, it would be nice to have an additional `git-sign-commit` boolean npm config flag which enabled `git commit -S` when creating the commit to save the new package version.

This would be a non-breaking feature addition since the default behavior could fallback to the current `git commit` when the `sign-git-commit=true` config is not present.

The change could me made [similar to this](https://github.com/npm/npm/blob/2a5977e0c65b244e92d848fcd56f2f80ba8cdf3b/lib/version.js#L261).  However, if I'm not mistaken, the `-S` flag in the `git commit` command may need to be spaced separately from the `-m` instead of being combined like the the `-sm` in the `git tag` command.

#### What problem is the feature intended to solve?
Enable consistent signing across both git tags and commits

#### Is the absence of this feature blocking you or your team? If so, how?
Not blocking

#### Is this feature similar to an existing feature in another tool?
Unknown

#### Is this a feature you're prepared to implement, with support from the npm CLI team?
My previous PR (#12697) is now out of date.  It probably needs to be rewritten.

**Comment by maxkoryukov** [comment]
ping (to prevent closing)

**Comment by tribou** [comment]
An positive UX update for this feature, the new version of `gpg-agent` now caches a password for 600 seconds by default.  Previously, this feature would require entering your password twice to sign for git (once for the commit, once for the tag).  However, the new default `gpg-agent` caching means users will only need to enter their GPG password once to apply to both commit and tag signings.

**Comment by maxkoryukov** [comment]
ping (to prevent closing)

**Comment by maxkoryukov** [comment]
ping (to prevent closing by bot)

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )

**Comment by legodude17** [comment]
@tribou It would be great if you could turn this into an RFC over at https://github.com/npm/rfcs!

**Comment by tribou** [comment]
Ah, that's neat! Didn't realize they started that. 👍 

Although my old PR #12697 is still open and was approved for merging in March, so I'm not sure if that needs an RFC? The PR just needs to be rewritten for npm@6.

**Comment by legodude17** [comment]
Yeah I didn't notice that. You could probably just rewrite it for npm@6.

**Comment by maxkoryukov** [comment]
ping (to prevent closing by @npm-robot )
